<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PathFinderClass</class>
 <widget class="QMainWindow" name="PathFinderClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>400</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>550</width>
    <height>400</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>550</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PathFinder</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QFrame" name="mazeViewPort">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QFrame" name="mazeFrame">
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0" alignment="Qt::AlignmentFlag::AlignHCenter|Qt::AlignmentFlag::AlignVCenter">
           <widget class="QTableWidget" name="mazeTable">
            <property name="styleSheet">
             <string notr="true">QTableView {
	gridline-color: rgba(255, 255, 255, 0);
}
QTableWidget::item {
	background-color: rgb(97, 97, 97);
}
QTableWidget::item:selected {
	background-color: rgb(85, 170, 127);
}
QTableWidget::item:focus {
	border: 1px solid yellow;
}</string>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="sizeAdjustPolicy">
             <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::MultiSelection</enum>
            </property>
            <property name="cornerButtonEnabled">
             <bool>false</bool>
            </property>
            <property name="rowCount">
             <number>4</number>
            </property>
            <property name="columnCount">
             <number>4</number>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderMinimumSectionSize">
             <number>10</number>
            </attribute>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>38</number>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderMinimumSectionSize">
             <number>10</number>
            </attribute>
            <attribute name="verticalHeaderDefaultSectionSize">
             <number>38</number>
            </attribute>
            <row/>
            <row/>
            <row/>
            <row/>
            <column/>
            <column/>
            <column/>
            <column/>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="resultsLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Path%1 found. Time Taken: %2 nanoseconds</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="optionsPane">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="optionsGroupBox">
         <property name="title">
          <string>Options</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="rowCountLabel">
            <property name="text">
             <string>Rows: 4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="rowSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>7</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="columnCountLabel">
            <property name="text">
             <string>Columns: 4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="columnSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>7</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="algorithmLabel">
            <property name="text">
             <string>Algorithm:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="algorithmComboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>115</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>Backtracking</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Branch &amp; Bound</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="markAsLabel">
            <property name="text">
             <string>Mark as:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="markSourceButton">
            <property name="text">
             <string>Source</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="markDestinationButton">
            <property name="text">
             <string>Destination</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item alignment="Qt::AlignmentFlag::AlignVCenter">
        <widget class="QWidget" name="actionButtonPane" native="true">
         <layout class="QVBoxLayout" name="VBoxLayout">
          <item>
           <widget class="QPushButton" name="findPathButton">
            <property name="text">
             <string>Find Path</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resetMazeButton">
            <property name="text">
             <string>Reset Maze</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="pathfinder.qrc"/>
 </resources>
 <connections/>
</ui>
